buffer/src/main/java/io/netty/buffer/api/MemoryManager.java

```diff
@@ -170,4 +189,14 @@ Buffer allocateShared(AllocatorControl allocatorControl, long size, Drop<Buffer>
      * @return The name of this memory managers implementation.
      */
     String implementationName();
+
+    /**
+     * A callback used to uninstall object leak callbacks that were previously
+     * {@linkplain #onLeakDetected(Consumer) installed}.
+     */
+    @UnstableApi
+    interface LeakCallbackUninstall extends AutoCloseable {
```

Comment:
I wonder if we should just move this interface somewhere else and call it `NonThrowingAutoClosable` or something like this. It might be general purpose later on.